src/common/config.h

```diff
@@ -634,6 +638,133 @@ class Config {
+    }
+    TryEnqueueFallback(mc_queue,
+                       event);
+    if (frame_id == this->frame_to_profile_) {
+      enqueue_end_tsc_ = GetTime::WorkerRdtsc();
+      enqueue_stats_[symbol_id][enqueue_stats_id_.at(symbol_id)].tsc_start_ =
+        enqueue_start_tsc_;
+      enqueue_stats_[symbol_id][enqueue_stats_id_.at(symbol_id)].tsc_end_ =
+        enqueue_end_tsc_;
+      enqueue_stats_[symbol_id][enqueue_stats_id_.at(symbol_id)].event_type_ =
+        event.event_type_;
+      enqueue_stats_id_.at(symbol_id)++;
+    }
+  }
+
+  inline void LogDequeueStatsMaster(EventType event_type,
+                                    size_t frame_id,
+                                    size_t tsc_dequeue_start,
+                                    size_t tsc_dequeue_end) {
+    dequeue_stats_[dequeue_stats_id_].tsc_start_ = tsc_dequeue_start;
+    dequeue_stats_[dequeue_stats_id_].tsc_end_ = tsc_dequeue_end;
+    dequeue_stats_[dequeue_stats_id_].event_type_ = event_type;
+    dequeue_stats_id_++;
+  }
+
+  inline void UpdateDequeueTscWorker(int tid,
+                                     size_t frame_id,
+                                     size_t dequeue_tsc,
+                                     size_t valid_dequeue_tsc) {
+    total_worker_dequeue_tsc_[tid][frame_id] += dequeue_tsc;
+    total_worker_valid_dequeue_tsc_[tid][frame_id] += valid_dequeue_tsc;
+  }
+
+  inline void UpdateEnqueueTscWorker(int tid,
+                                     size_t frame_id,
+                                     size_t enqueue_tsc) {
+    total_worker_enqueue_tsc_[tid][frame_id] += enqueue_tsc;
+    worker_num_valid_enqueue_[tid][frame_id]++;
+  }
+
+  inline void LogEnqueueStatsWorker(int tid,
+                                    size_t symbol_id,
+                                    size_t start_tsc,
+                                    size_t end_tsc,
+                                    EventType event_type) {
+    size_t id = worker_enqueue_stats_id_[tid][symbol_id];
+    worker_enqueue_stats_[tid][symbol_id][id].tsc_start_ = start_tsc;
+    worker_enqueue_stats_[tid][symbol_id][id].tsc_end_ = end_tsc;
+    worker_enqueue_stats_[tid][symbol_id][id].event_type_ = event_type;
+    worker_enqueue_stats_id_[tid][symbol_id]++;
+  }
+
+  inline void LogDequeueStatsWorker(int tid,
+                                    size_t symbol_id,
+                                    size_t start_tsc,
+                                    size_t end_tsc,
+                                    EventType event_type) {
+    size_t id = worker_dequeue_stats_id_[tid][symbol_id];
+    worker_dequeue_stats_[tid][symbol_id][id].tsc_start_ = start_tsc;
+    worker_dequeue_stats_[tid][symbol_id][id].tsc_end_ = end_tsc;
+    worker_dequeue_stats_[tid][symbol_id][id].event_type_ = event_type;
+    worker_dequeue_stats_id_[tid][symbol_id]++;
+  }
+
+  // Task enqueue/dequeue start and end timestamps and task type
+  struct QueueTsStat {
```

Comment:
try not to bury struct / class definitions in a specif class definition file.  Refer to the coding standard for correct approaches.

Reply:
Also, Config does not seem like a relevant place for these functions and parameters. Why can't these be in Stats class or in Doer?